<?php
echo "Начало программы\n";

try {
    // Код, в котором перехватываются исключения
    echo "Все, что имеет начало...\n";
    
    // Генерируем ("выбрасываем") исключение
    throw new Exception("Случилось страшное!", 100);

    echo "...имеет и конец";
} catch (Exception $exp) {
    // Код обработчика
    echo " Исключение: {$exp->getMessage()}, код: {$exp->getCode()}\n";
}

echo "Конец программы\n";
